pkg/sdk/secrets_gen_test.go

```diff
@@ -0,0 +1,321 @@
+package sdk
+
+import "testing"
+
+func TestSecrets_CreateWithOAuthClientCredentialsFlow(t *testing.T) {
+	id := randomSchemaObjectIdentifier()
+
+	defaultOpts := func() *CreateWithOAuthClientCredentialsFlowSecretOptions {
+		return &CreateWithOAuthClientCredentialsFlowSecretOptions{
+			name: id,
+		}
+	}
+
+	t.Run("validation: nil options", func(t *testing.T) {
+		var opts *CreateWithOAuthClientCredentialsFlowSecretOptions = nil
+		assertOptsInvalidJoinedErrors(t, opts, ErrNilOptions)
+	})
+	t.Run("validation: invalid identifier", func(t *testing.T) {
+		opts := defaultOpts()
+		opts.name = emptySchemaObjectIdentifier
+		assertOptsInvalidJoinedErrors(t, opts, ErrInvalidObjectIdentifier)
+	})
+
+	t.Run("validation: conflicting fields for [opts.OrReplace opts.IfNotExists]", func(t *testing.T) {
+		opts := defaultOpts()
+		opts.IfNotExists = Bool(true)
+		opts.OrReplace = Bool(true)
+		assertOptsInvalidJoinedErrors(t, opts, errOneOf("CreateWithOAuthClientCredentialsFlowSecretOptions", "OrReplace", "IfNotExists"))
+	})
+
+	t.Run("all options", func(t *testing.T) {
+		integration := randomAccountObjectIdentifier()
+
+		opts := defaultOpts()
+		opts.IfNotExists = Bool(true)
+		opts.SecurityIntegration = integration
+		opts.OauthScopes = []SecurityIntegrationScope{{"test"}}
```

Comment:
If it's possible in Snowflake, there should be a test with an empty list.

Reply:
Have made one, but the SQL builder seems to omit the `OAUTH_SCOPES` when provided with empty list `[]SecurityIntegrationScope{}`